tests/PrefixCacheBootstrapperTest.php

```diff
@@ -0,0 +1,345 @@
+    $prefixBase = 'custom_';
+
+    config([
+        'tenancy.cache.prefix_base' => $prefixBase
+    ]);
+
+    $tenant1 = Tenant::create();
+    tenancy()->initialize($tenant1);
+
+    expect($originalPrefix . $prefixBase . $tenant1->getTenantKey() . ':')
+        ->toBe(app('cache')->getPrefix())
+        ->toBe(app('cache.store')->getPrefix());
+});
+
+test('cache is prefixed correctly when using a repository injected in a singleton', function () {
+    $this->app->singleton(CacheService::class);
+
+    $this->app->make(CacheService::class)->handle();
+
+    expect(cache('key'))->toBe('central-value');
+
+    $tenant1 = Tenant::create();
+    $tenant2 = Tenant::create();
+    tenancy()->initialize($tenant1);
+
+    expect(cache('key'))->toBeNull();
+    $this->app->make(CacheService::class)->handle();
+    expect(cache('key'))->toBe($tenant1->getTenantKey());
+
+    tenancy()->initialize($tenant2);
+
+    expect(cache('key'))->toBeNull();
+    $this->app->make(CacheService::class)->handle();
+    expect(cache('key'))->toBe($tenant2->getTenantKey());
+
+    tenancy()->end();
+
+    expect(cache('key'))->toBe('central-value');
+});
+
+test('specific central cache store can be used inside a service', function () {
+    config(['cache.default' => 'redis']);
+    config(['cache.stores.redis2' => config('cache.stores.redis')]);
+    $cacheStore = 'redis2'; // Name of the non-default, central cache store that we'll use using cache()->store($cacheStore)
+
+    // Service uses the 'redis2' store which is central/not prefixed (not present in PrefixCacheTenancyBootstrapper::$tenantCacheStores)
+    $this->app->singleton(SpecificCacheStoreService::class, function() use ($cacheStore) {
+        return new SpecificCacheStoreService($this->app->make(CacheManager::class), $cacheStore);
+    });
+
+    $this->app->make(SpecificCacheStoreService::class)->handle();
+    expect(cache()->store($cacheStore)->get('key'))->toBe('central-value');
+
+    $tenant1 = Tenant::create();
+    $tenant2 = Tenant::create();
+    tenancy()->initialize($tenant1);
+
+    // The store isn't prefixed, so the cache isn't separated
+    expect(cache()->store($cacheStore)->get('key'))->toBe('central-value');
+    $this->app->make(SpecificCacheStoreService::class)->handle();
+    expect(cache()->store($cacheStore)->get('key'))->toBe($tenant1->getTenantKey());
+
+    tenancy()->initialize($tenant2);
+
+    expect(cache()->store($cacheStore)->get('key'))->toBe($tenant1->getTenantKey());
+    $this->app->make(SpecificCacheStoreService::class)->handle();
+    expect(cache()->store($cacheStore)->get('key'))->toBe($tenant2->getTenantKey());
+
+    tenancy()->end();
+    expect(cache()->store($cacheStore)->get('key'))->toBe($tenant2->getTenantKey());
+});
+
+test('stores specified in tenantCacheStores get prefixed', function() {
+    // Make the currently used store ('redis') the only store in $tenantCacheStores
+    PrefixCacheTenancyBootstrapper::$tenantCacheStores = ['redis'];
```

Comment:
How does this test assert that this is happening specifically with the `redis` cache store?

Reply:
The test seems pretty similar to the other tests above. Does it do anything extra? Or is it just present as a "contrast" to the next test (`stores not specified in tenantCacheStores do not get prefixed`)?

If that's the case, maybe you could merge those into one test that's named something like `only stores specified in tenantCacheStores get prefixed` where you'd assert that the specified stores do get prefixed, and the ones not specified don't.

Reply:
> is it just present as a "contrast" to the next test

That's right

> maybe you could merge those into one test that's named something like only stores specified in tenantCacheStores get prefixed where you'd assert that the specified stores do get prefixed, and the ones not specified don't

Okay, that makes sense. I'll merge them

Reply:
Something like this should do it: https://github.com/archtechx/tenancy/pull/1014/commits/4c9cb7cf51b9d3002b61b567d8df39a34d4ad7cb

Maybe I'd extract the code into the closure to make the test cleaner/reduce duplicity